modules/library/src/main/scala/zio/elasticsearch/query/package.scala

```diff
@@ -120,6 +120,21 @@ package object query {
     def innerHits(innerHits: InnerHits): Q
   }
 
+  private[elasticsearch] trait HasMinimumShouldMatch[Q <: HasBoost[Q]] {
+
+    /**
+     * Sets the `minimumShouldMatch` parameter for this [[ElasticQuery]]. The `minimumShouldMatch` value is the number
+     * of should clauses returned documents must match. If the [[zio.elasticsearch.query.BoolQuery]] includes at least
+     * one `should` clause and no `must`/`filter` clauses, the default value is 1. Otherwise, the default value is 0.
+     *
+     * @param value
+     *   a number to set `minimumShouldMatch` parameter to
+     * @return
+     *   a new instance of the [[ElasticQuery]] with the `minimumShouldMatch` value set.
```

Comment:
If this applies only on `BoolQuery`, maybe write it here instead of `ElasticQuery`.

Reply:
Not sure if there are any other places to be supported.

Reply:
At least [match bool prefix](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/query-dsl-match-bool-prefix-query.html) query supports this. So, let's have it in a separate trait.